DigitalLearningSolutions.Data.Tests/Services/UserDataServiceTests.cs

```diff
@@ -60,7 +60,7 @@ public void GetAdminUserById_Returns_admin_user()
         public void GetDelegateUserById_Returns_delegate_users()
         {
             // Given
-            var expectedDelegateUsers = UserTestHelper.GetDefaultDelegateUser();
+            var expectedDelegateUsers = UserTestHelper.GetDefaultDelegateUser(jobGroupName: "Nursing / midwifery");
```

Comment:
If the default delegate user should have a job group name of "Nursing / midwifery" I think it should be put in the GetDefaultDelegateUser default options. 

I suspect that might mess up other tests - e.g. some data service tests will use queries that don't recover all the columns (as they won't get used further down the line), and then compare if the default model is equivalent to the recovered data (which will have nulls for those missed columns). So we might want a decision for how to move forward. I see three options:

1. Change all the queries that retrieve data to always retrieve ALL the columns, and change the GetDefault...() methods to match all the columns on the DB record they are for as the defaults.
2. Don't change the queries that retrieve data, but change the GetDefault...() as in number 1. This would mean each test would have to specify the null columns to overwrite the defaults in GetDefault...() so there would be a lot of going back and forth "correcting" tests when new columns are added.
3. Keep it as it is in this file, so queries only retrieve what is necessary, but then we have to specify in the tests what the values the expected one should have.

My preference is for option 1, though it may require a large amount of overhauling depending on how many data services don't get the full set of columns for their tables. It would probably entail us doing what we are doing for the data services split - let everyone know, and then when they work on a service they can change the rest of it to match, so we only change one service at a time. Thoughts?

Reply:
You are exactly right, it was because of a failing unit test I did it this way, one query not having all the properties that the other did. I do think all relevant queries should probably be populating all the properties on the. e.g. all AdminUser queries should return fully populated AdminUsers. Does bring us back around to the JobGroupService question then. Should we populate the ID and look it up, or add the Join to every query?

Reply:
I think if we are deciding to make all queries populate all properties then we should use JobGroupService for the acquiring of the name (and thus, keeping with the same principal, do the same for the CentreName). 

Though I am not sure if we should then be adding CentreName/JobGroupName to the data models or keep them separate. If we don't add them, then the data model and data service will be clear matches to the database, but we will likely have to refactor other services/controllers to obtain the names properly. If we add them to the model (but have default values as null or empty), then we can easily refactor the service methods to make the calls to get the additional properties and leave most of the code the same, but then we don't have a model = database pattern.

Reply:
I think the "getAccountByUsername" only cares about the fields that are required for logging in whereas "byId" cares about all the information. Hence, I think joining is still a good idea and we shouldn't change the code because of this single test (unless we truly believe that we'll need these extra fields later down the line). 
In general, I think it's ok to override this field in the "default user", because the default user is meant to be the simplest user and is there to avoid some code duplication (kind of like a builder).